benchmarking/runs/social/varied_num_friends.py

```diff
@@ -0,0 +1,189 @@
+from typing import Dict, List
+
+import typer
+
+from api.ai.interfaces.algorithm_config import (
+    WeightAlgorithmConfig,
+    SocialAlgorithmConfig,
+    RandomAlgorithmConfig,
+    AlgorithmConfig,
+)
+from api.models.enums import AlgorithmType
+from benchmarking.data.simulated_data.mock_student_provider import (
+    MockStudentProviderSettings,
+    MockStudentProvider,
+)
+from benchmarking.evaluations.graphing.graph_metadata import GraphData, GraphAxisRange
+from benchmarking.evaluations.graphing.line_graph import line_graph
+from benchmarking.evaluations.graphing.line_graph_metadata import LineGraphMetadata
+from benchmarking.evaluations.interfaces import TeamSetMetric
+from benchmarking.evaluations.metrics.average_social_satisfied import (
+    AverageSocialSatisfaction,
+)
+from benchmarking.evaluations.metrics.utils.team_calculations import *
+from benchmarking.evaluations.scenarios.give_the_people_what_they_want import (
+    GiveThePeopleWhatTheyWant,
+)
+from benchmarking.runs.interfaces import Run
+from benchmarking.simulation.insight import Insight, InsightOutput
+from benchmarking.simulation.simulation_set import SimulationSet, SimulationSetArtifact
+from benchmarking.simulation.simulation_settings import SimulationSettings
+
+
+class VariedNumFriendsSocialRun(Run):
+    @staticmethod
+    def start(num_trials: int = 1, generate_graphs: bool = True):
+        """
+        Goal: See how the social algorithm reacts to different numbers of friends.
+
+        This run holds the team size constant at 4, then tests it with cliques that are 0.5, 1,
+        1.5, and 2 times the size of the team. This will test to see if the algorithm will be
+        able to put cliques into teams if the clique size does not match the team size.
+        The most interesting case will be the 0.5 case because it should be able to put two
+        full cliques into a single team.
+        """
+
+        clique_sizes = [2, 4, 6, 8]
+        class_sizes = [100, 200]  # list(range(50, 501, 50))
+
+        metrics: Dict[str, TeamSetMetric] = {
```

Comment:
same comment as with seth's thing, you could extend Run into some kind of SocialRun or whatever and put all the repeated code in methods there